### PR TITLE
feat: enhance render confirmation dialog with parameter comparison

### PR DESCRIPTION
**Pull request by mhuang74**
## Summary

When a user clicks "Start Render" and a previous successful render exists, show a **side-by-side comparison table** of the previous render job's parameters vs. the current requested parameters, with **visual diff highlighting** for changed values.

### `webapp/src/lib/constants.ts`
- Added `TEMPLATES`, `RESOLUTIONS`, and `FONT_SIZES` exports alongside the existing `FONT_FAMILIES`
- These were previously local to `RenderForm.tsx` and duplicated in `SettingsForm.tsx`

### `webapp/src/db/schema.ts`
- Added `songCount` (integer, nullable) and `songsetDurationSeconds` (integer, nullable) columns to `renderJobs` table
- These are snapshots captured at render creation time for comparison

### `webapp/drizzle/0013_plain_starfox.sql`
- Migration: `ALTER TABLE "render_jobs" ADD COLUMN "song_count" integer; ALTER TABLE "render_jobs" ADD COLUMN "songset_duration_seconds" integer;`

### `webapp/src/lib/render/job-manager.ts`
- `createRenderJob()` now queries `songsetItems` with `recordings.durationSeconds` to compute `songCount` and `songsetDurationSeconds` at render creation time
- `RenderJob` interface updated with `songCount: number | null` and `songsetDurationSeconds: number | null`
- `mapRowToRenderJob()` maps the new columns

### `webapp/src/components/render/RenderForm.tsx`
- Imported `FONT_SIZES`, `TEMPLATES`, `RESOLUTIONS` from `@/lib/constants` (removed local definitions)
- Imported `formatDuration` from `@/lib/format`
- Extracted `PreviousRenderJobData` as a named exported interface (was inline type with 7 fields, now 10+)
- Added `currentSongCount` and `currentSongsetDurationSeconds` props
- Added `formatDurationSafe()` helper — null-guards `formatDuration()` calls
- Added `isDifferent()` helper — compares previous vs current values with null normalization and numeric rounding
- Replaced simple parameter list in confirmation dialog with a 3-column comparison table (Parameter / Previous Render / Current Request)
- Changed values get amber text + dot indicator
- Estimated total duration shows `~` prefix (songset duration + title card if enabled)
- Null/missing values display `—`

### `webapp/src/components/settings/SettingsForm.tsx`
- Imported `TEMPLATES`, `RESOLUTIONS` from `@/lib/constants` (removed local definitions)

### `webapp/src/app/songsets/[id]/render/page.tsx`
- Extended `SongsetData` with `durationSeconds: number | null`
- Extended `RenderJobData` with `resolution`, `totalDurationSeconds`, `songCount`, `songsetDurationSeconds`
- Passes `currentSongCount` and `currentSongsetDurationSeconds` props to `RenderForm`
- Passes new fields (`resolution`, `totalDurationSeconds`, `songCount`, `songsetDurationSeconds`) in `previousRenderJob` prop

### `webapp/src/test/components/render/RenderForm.test.tsx`
- Added 8 new test cases for the comparison dialog:
  1. Shows comparison dialog when previousRenderJob is provided
  2. Diff highlighting for changed values
  3. No diff highlighting for same values
  4. Shows `—` for null previous fields
  5. Submits directly without dialog when no previous render
  6. Displays current song count and duration correctly
  7. Estimated total duration shows `~` prefix with title card
  8. `formatDurationSafe` returns `—` for null/undefined/zero

### `webapp/src/test/db/schema.test.ts`
- Added test for `song_count` and `songset_duration_seconds` columns on `renderJobs`

### `webapp/src/test/lib/render/job-manager.test.ts`
- Updated DB mock to support `leftJoin` chain method
- Added `songCount` and `songsetDurationSeconds` to mock render job

## Edge Cases Handled

- **Historical render jobs** (pre-migration): `songCount` and `songsetDurationSeconds` are `null` → display `—`
- **Null `totalDurationSeconds`**: Worker may not have set it → display `—` via `formatDurationSafe()`
- **Float precision**: `isDifferent()` rounds numeric comparisons; `songsetDurationSeconds` uses `integer` column type
- **Null safety**: `formatDurationSafe()` guards against null/undefined/zero before calling `formatDuration()`

## Testing

- All 1361 tests pass (including 8 new comparison dialog tests)
- `pnpm lint` passes with 0 errors
- DB migration generated and pushed to dev database

### CODE REVIEW

File: webapp/src/components/render/RenderForm.tsx [hunk 1]
```diff
@@ -482,26 +491,63 @@ export function RenderForm({
...
+                    const currTotalDuration = estimatedTotalDuration > 0 ? `~${formatDuration(estimatedTotalDuration)}` : "—"
+
+                    const rows: { label: string; prev: string; curr: string; diff: boolean }[] = [
+                      { label: "Font", prev: prevFont, curr: currFont, diff: isDifferent(previousRenderJob.fontFamily, formData.fontFamily) },
+                      { label: "Font Size", prev: prevFontSize, curr: currFontSize, diff: isDifferent(previousRenderJob.fontSizePreset, formData.fontSizePreset) },
+                      { label: "Background", prev: prevTemplate, curr: currTemplate, diff: isDifferent(previousRenderJob.template, formData.template) },
+                      { label: "Resolution", prev: prevResolution, curr: currResolution, diff: isDifferent(previousRenderJob.resolution, formData.resolution) },
+                      { label: "Title Card", prev: prevTitleCard, curr: currTitleCard, diff: isDifferent(previousRenderJob.includeTitleCard, formData.includeTitleCard) || isDifferent(previousRenderJob.titleCardDurationSeconds, formData.titleCardDurationSeconds) },
```
[the remark this answers is not in the document]
Comment (mhuang74) [reply webapp/src/components/render/RenderForm.tsx]: Fixed in a8e6eaf. Now only compares titleCardDurationSeconds when includeTitleCard is true, using ```?? 10``` fallback for the previous value to match the display logic.

File: webapp/src/components/render/RenderForm.tsx [hunk 2]
```diff
@@ -482,26 +491,63 @@ export function RenderForm({
...
+                      { label: "Songs", prev: prevSongCount, curr: currSongCount, diff: isDifferent(previousRenderJob.songCount, currentSongCount) },
+                      { label: "Songset Duration", prev: prevSongsetDuration, curr: currSongsetDuration, diff: isDifferent(previousRenderJob.songsetDurationSeconds, currentSongsetDurationSeconds) },
+                      { label: "Total Duration", prev: prevTotalDuration, curr: currTotalDuration, diff: isDifferent(previousRenderJob.totalDurationSeconds, estimatedTotalDuration) },
```
[the remark this answers is not in the document]
Comment (mhuang74) [reply webapp/src/components/render/RenderForm.tsx]: Fixed in a8e6eaf. Now normalizes 0 to null using `|| null` for both songsetDurationSeconds and totalDurationSeconds comparisons, so null and 0 are treated as equivalent (both display as "—").
